deep_water_level/src/data.py

```diff
@@ -46,11 +46,14 @@ def get_data_loaders(
     batch_size: int = 32,
     train_test_split: Tuple[int, int] = [0.8, 0.2],
 ):
-    # Define a transform to apply to the data
-    transform = transforms.Compose([transforms.ToTensor()])
+    transforms = v2.Compose(
+    [
+        v2.ToImage(),
+        v2.ToDtype(torch.float32, scale=True), # convert to float32 and normalize to 0, 1
```

Comment:
how does it know how to normalize to 0, 1? Does it use the whole dataset to learn the range or does it make an assumption about it?

My recollection when I implemented Grounding Dino is that I had to pre-calculate the mean and stdev per channel from the dataset in order to normalize correctly. See https://github.com/jonbinney/deep_rabbit_hole/blob/49c35bac4d232497f06aec9df5fdeb025776c65a/object_tracker_0/src/inference_gd.py#L48

Maybe v2 does this automatically or something? do you know @jonbinney ?

Reply:
Oh, not that NN-specific kind of normalization. This is just scaling the values so that they are between 0 and 1. So for uint8 channels, dividing by 255.